src/helpers/axios.js

```diff
@@ -0,0 +1,29 @@
+import axios from 'axios';
+import { currentUser, refreshUser } from 'redux/auth/operations';
+import { store } from 'redux/store';
+
+export const API = axios.create({
+  baseURL: '',
+  withCredentials: true,
+});
+
+API.interceptors.response.use(
+  response => response,
+  async error => {
+    const originalRequest = error.config; // can use directly error.config._retry
+    if (error.response.status === 401 && !originalRequest._retry) {
+      originalRequest._retry = true;
+      try {
+        store.dispatch(refreshUser());
```

Comment:
please use await 
like this 
        await store.dispatch(refreshUser()).unwrap();
        await store.dispatch(currentUser()).unwrap();